docs/src/almalinux_instructions.rst

```diff
@@ -0,0 +1,248 @@
+
+.. note::
+    The ``carta`` user should not be in the ``carta-users`` group. ``carta-users`` should only be assigned to the normal user accounts.
+
+7. Set up the user authentication method
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This is the most difficult step and depends on how you authenticate users at your institute. In this step-by-step guide we use PAM local authentication and a local user, ``bob``, on the server running the CARTA controller. The user ``bob`` needs to be part of the ``carta-users`` group.
+
+With PAM authentication, the ``carta`` user that runs the CARTA controller requires access to the ``/etc/shadow`` file in order to authenticate other users. We can enable this by creating a new group called ``shadow`` and assigning the ``/etc/shadow`` file to that group.
+
+.. note::
+    Only PAM with local authentication requires ``/etc/shadow`` access. PAM using LDAP, and Google OAuth, do not require ``/etc/shadow`` access. 
+
+.. code-block:: shell
+
+    # Create the test user 'bob':
+    sudo useradd -G carta-users bob
+    sudo passwd bob
+
+    # A new group called 'shadow' needs to be assinged to the /etc/shadow file and user 'carta':
+    sudo groupadd shadow
+    sudo chgrp shadow /etc/shadow
+    sudo chmod g+r /etc/shadow
+    sudo usermod -a -G shadow carta
+    ls -l /etc/shadow
+    # It should show permissions as ----r-----. 1 root shadow
+    # It could be helpful to reboot the server at this point
+    sudo reboot 
+
+
+8. Configure the CARTA controller
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Create and fill in the ``config.json`` using our :ref:`sample configuration file<example_config>`. 
+Also generate private/public keys as they are used by the CARTA controller to sign/verify/refresh access tokens.
+
+.. code-block:: shell
+
+    sudo mkdir /etc/carta
+    sudo chown -R carta /etc/carta
+    vi /etc/carta/config.json
+    
+    # Generate private/public keys:
+    cd /etc/carta
+    sudo openssl genrsa -out carta_private.pem 4096
+    sudo openssl rsa -in carta_private.pem -outform PEM -pubout -out carta_public.pem
+
+Please check the `CARTA Configuration Schema <https://carta-controller.readthedocs.io/en/latest/schema.html#schema>`_ for all available options.
+
+
+9. Check everything is working
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Here we switch to the ``carta`` user and test the CARTA controller with our test user ``bob``:
+
+.. code-block:: shell
+
+    su - carta
+    carta-controller -verbose -test bob
+
+If the test is successful, the CARTA controller should be ready to deploy.
+
+
+10. Start the CARTA controller
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: shell
+
+    su - carta
+    carta-controller
+
+Now your users should be able to access your server's URL and log into CARTA.
+
+
+Optional: Set up the CARTA controller to run with pm2
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Comment:
This underline needs to be adjusted to match the new heading text (Sphinx warns about it).